### ISSUE TITLE
Tapping bar items should clear navigation stack

### ISSUE THREAD

**Issue by jlnr** [lurado/LDOBarController#30]
This is a built-in feature of `UITabBarController` when it contains `UINavigationController`s: Tapping the tab bar item will usually return to the root of the navigation stack.

Curious exception: In Photos.app, when iCloud Sharing is active (not iCloud Photo Library...), tapping the middle tab item will actually go to 'Activity', not to the root. Is there an official API to implement this? Can we use this API?

**Comment by sebastianludwig** [comment]
How about checking for a child `UINavigationController` and simply calling `popToRootViewController`?

**Comment by jlnr** [comment]
Yep, the only thing we need to be careful about is if this happens when tapping, or when releasing the bar item.
